### DISCUSSION TITLE
Writing text to image -- issue with types

### DISCUSSION THREAD

**Discussion by lokijota** [dlemstra/Magick.NET#708] [Help]
Hi,

I've read of two different ways to write text t images. One is this:

```csharp
new Drawables()
.FontPointSize(20)
.Font("Arial", FontStyleType.Normal, FontWeight.Bold, FontStretch.Normal)
.StrokeColor(MagickColors.Red)
.FillColor(MagickColors.White)
.TextAlignment(TextAlignment.Left)
.Text(10, 10, text)
.Draw(image);
```

This is not working for me, because Draw expects an `IMagickImage<ushort>` and my `image` is `IMagickImage`.

The other one is listed in the documentation here: <https://github.com/dlemstra/Magick.NET/blob/master/docs/Drawing.md>. This second one uses:

`image.Composite(caption, 590, 450, CompositeOperator.Over);`

But the `IMagickImage` interface doesn't have the `Composite()` method.

I'm a bit lost with `MagickImage`, `IMagickImage`, `IMagickImage<ushort>` and the fact that they seem to have diffferent capabilities and be accepted by different methods. Is there any documentation page to help understand this? and how do I write to my `IMagickImage`?

Thanks!

**Comment by dlemstra** [comment]
If you need those methods you should change the use `IMagickImage` of to `IMagickImage<ushort>`. The `IMagickImage<ushort>` inherits from the `IMagickImage` interface and contain operations that need to know the Quantum type. It turns out that in this case the `Composite` method does not require it because none of the other arguments need to know the Quantum type so I will need to change that in a future release. But the `Draw` method needs this because `StrokeColor` needs to know the Quantum type. Now also wondering if I could change that in the future and allow passing in `IMagickImage` to the `Draw` method. The `IMagickImage<ushort>` interface was introduced to make it possible to add extension methods to the `IMagickImage` interface for methods that are unaware of the `QuantumType`.

Long story short. You are probably doing something along the lines of `IMagickImage image = new MagickImage()` and that should be changed to `IMagickImage<ushort> image = new MagickImage()`.
